docs/src/modules/java/pages/event-sourced-entities.adoc

```diff
@@ -179,6 +179,32 @@ When the Event Sourced Entity is loaded again, the snapshot will be loaded befor
 
+=== Replication filters
+
+Events are by default replicated to all regions that have been enabled for the service. For regulatory reasons or as cost optimization it is possible to filter which regions that participate in the replication for a specific entity. This can be changed at runtime by the entity itself.
+
+[source,java]
+.ShoppingCartEntity.java
+----
+include::example$doc-snippets/src/main/java/com/example/replicationfilter/ShoppingCartEntity.java[tag=replication-filter]
+----
+<1> Enable the replication filter feature by adding the `@EnableReplicationFilter` annotation.
+<2> Define the replication filter with the `updateReplicationFilter` effect.
+
+After enabling the replication filter the entity is still replicated to all regions until the regions are defined with the `updateReplicationFilter` effect. This effect can be combined with persisting events or used without additional events. The filter can only be updated from the entity's primary region. With the `request-region` primary selection strategy, updating the filter from a non-primary region will cause that region to become the new primary. The filter is durable for the specific entity instance and can be changed without deploying a new version.
+
+In the link:_attachments/api/akka/javasdk/eventsourcedentity/ReplicationFilter.html[`ReplicationFilter`] you define the regions to be included or excluded in the replication. The region where the update is made, the so-called self region, is automatically included in the replication filter and cannot be excluded. The changes are additive for each entity instance, meaning that if you first `updateReplicationFilter` and include `gcp-us-east1` and then later make another `updateReplicationFilter` and include `aws-us-east-2` from the same entity, then both `gcp-us-east1` and `aws-us-east-2` are included.
+
+When you add the `@EnableReplicationFilter` annotation the entity will still replicate to all regions until you have defined a filter with `updateReplicationFilter`. You can define the filter when persisting events, including the first event. For example, this is how to effectively disable replication to other regions for a specific entity by defining a filter that only includes the self region:
+
+[source,java]
+.ShoppingCartEntity.java
+----
+include::example$doc-snippets/src/main/java/com/example/replicationfilter/ShoppingCartEntity.java[tag=replication-filter-event]
+----
+
+If you start with an entity with such self region filter in `gcp-us-east1`, and then later receive a command (not read-only) for this entity instance in another `aws-us-east-2`, it will automatically synchronize all events in from `gcp-us-east1` before handling the command in `gcp-us-east1`. Such command will also automatically include `aws-us-east-2` to the replication filter and events will be replicated to both `gcp-us-east1` and `aws-us-east-2`. You can remove a region from the replication with `ReplicationFilter.excludeRegion`.
+
```

Comment:
I was thinking about how we can handle replication filter for the agent session memory. Perhaps we can always make the session memory self region only, as described above? If the session is used from another region it will sync and automatically add that region in the filter.